### ISSUE TITLE
Render highway=passing_place

### ISSUE THREAD

**Issue by boothym** [gravitystorm/openstreetmap-carto#2356]
Over 12,000 uses as a node: https://wiki.openstreetmap.org/wiki/Tag:highway%3Dpassing_place

Plenty of them in the Scottish Highlands, example: https://goo.gl/maps/vriCnHkdvCr

Perhaps it can be rendered as a bulge in the road, similar to a turning circle but longer and narrower? Then in future it could be rendered on just one side of the road, if the tagging is updated (e.g. passing_place=left/right/both).

**Comment by SomeoneElseOSM** [comment]
"highway=passing_place" tends to be tagged on a node, with no idea of "right" or "left".  The approach I took in a related style was simply to render in the same way as turning circles:

![10102](https://cloud.githubusercontent.com/assets/5311740/18651646/72266aa4-7ec4-11e6-9843-c920b703e103.png)

It's a bit stylized, but gets the point across.  That tile is at http://www.openstreetmap.org/#map=16/56.6191/-4.9298 , just down the road from your Google Streetview link.

**Comment by boothym** [comment]
That looks ok, although maybe on thinner roads in this style the turning circle might look too big. Really need to see it rendered and also compared with a stretched version as well.

**Comment by dieterdreist** [comment]
2016-09-21 0:46 GMT+02:00 boothym <notifications@github.com>:

> That looks ok, although maybe on thinner roads in this style the turning
> circle might look too big. Really need to see it rendered and also compared
> with a stretched version as well.
>


this looks way too important for this kind of feature (IMHO). Isn't this
about the amount of lanes? We don't show lanes in general in this style.

**Comment by SomeoneElseOSM** [comment]
@dieterdreist No, this is not "about the amount of lanes".  It's about displaying the only place where two wide vehicles can pass each other, possibly for miles.

You could argue that this style is "only designed for cities", and while some changes clearly reflect that (see the discussion on https://github.com/gravitystorm/openstreetmap-carto/pull/747 for example) I don't think anyone has been quite so explicit before.

**Comment by dieterdreist** [comment]
sent from a phone

> Il giorno 21 set 2016, alle ore 10:19, SomeoneElseOSM <notifications@github.com> ha scritto:
> 
> @dieterdreist No, this is not "about the amount of lanes". It's about displaying the only place where two wide vehicles can pass each other, possibly for miles.
> 


so if they meet in the middle they have to agree who will backup several miles with his wide vehicle? Or are they typically leaving the carriageway to resolve?


> You could argue that this style is "only designed for cities", and while some changes clearly reflect that (see the discussion on #747 for example) I don't think anyone has been quite so explicit before.
> 

while I agree that it works better for dense areas, it should still be somehow usable also in rural settings, agreed. 
I was commenting on the visual representation, IMHO those "blobs" are too obtrusive, if this is a rare but important situation, maybe a small unobtrusive signature (icon) would be more suitable (something as small and colourless as the gate icon)? Or a thickening, but much less than the one given as example

**Comment by jeisenbe** [comment]
The documentation for this feature is not so clear. While the description in the infobox says "A passing place on a single track road", the rest of the page suggest it could be used on wider roads:

"Location of a widening on a road allowing oncoming vehicles to pass each other, or allowing slower traffic to be passed if said slower vehicle halts at the passing place. Also known as a turnout."

As noted above, there is no indication about which side of the road has the turnout/passing place, so we cannot render it in a very helpful way. 

Since we do not render any indication of lanes or width of roads, or places where overtaking is allowed, I don't think we can show this feature in a reasonable way.

**Comment by SomeoneElseOSM** [comment]
> The documentation for this feature is not so clear. While the description in the infobox says "A passing place on a single track road", the rest of the page suggest it could be used on wider roads:

Usage, at least wherever I've looked in OSM, seems to be consistent.

> As noted above, there is no indication about which side of the road has the turnout/passing place, so we cannot render it in a very helpful way.

That's simply not true, as I showed back in 2016.

By all means say "we don't want to render this now because we don't have the develop resource" or "because it doesn't it doesn't fit with the road features that we currently show" or "we believe that the suggested rendering will confuse users without a map key, which we don't have", but please don't say that you _can't_ render it, when an example of how you _can_ has already been given.

**Comment by jeisenbe** [comment]
I appologise for being unclear. When I wrote "we cannot render it in a
very helpful way" I did not mean it was impossible to render this
feature, but that the rendering possibilities would not be helpful to
general map users, and might be confusing.

If someone wants to submit a PR or suggest a different possible
rendering, that is still possible.

(While there is a key to this style, it is buried in the
Openstreetmap.org wiki, and not widely known.)

**Comment by polarbearing** [comment]
Given the feature being in the same category as a turning circle I find the proposed bulge not confusing at all. How did you test that this option is "not ... helpful to general map users"?

Please re-open.

**Comment by boothym** [comment]
>the rest of the page suggest it could be used on wider roads:

Does it? It says **"allowing oncoming vehicles to pass each other"** - on a "normal" width road passing places don't exist because they aren't needed, as there is enough room for cars to pass each other in two way traffic. 

There is a secondary function of allowing slower vehicles (tractors etc) to pull off to the side and allow cars past.

>The documentation for this feature is not so clear. 

Well, we can easily fix the documentation. 😉 

> would not be helpful to general map users, and might be confusing.

Could you explain how it might be confusing?

This feature is now used 22,500 times in OSM - an increase of 10k since I posted this 3 and a half years ago.

**Comment by pnorman** [comment]
I'd have no issues with a rendering like @SomeoneElseOSM showed

**Comment by jeisenbe** [comment]
Reopened. Is anyone interested in working on this? @SomeoneElseOSM?

**Comment by polarbearing** [comment]
30.6K

**Comment by leijurv** [comment]
This caught my interest, I made a prototype (based on the #5296 code). A cool side effect is that `surface` can be respected pretty easily in the passing place. In this comparison we can see how Chumash Highway is the same surface, whereas `surface=dirt` is shown on most of the passing places of Painted Cave Road.

[`#17/34.49048/-119.79013`](https://www.openstreetmap.org/#map=17/34.490480/-119.790130)

|Before|After|
|---|---|
|<img width="622" height="772" alt="Image" src="https://github.com/user-attachments/assets/5b4bc7b5-4c19-4291-b14d-ff3c21d4a8de" />|<img width="622" height="772" alt="Image" src="https://github.com/user-attachments/assets/11444b72-b392-43dd-8c94-b6ba0b35252a" />|

**Comment by dch0ph** [comment]
Looks promising! The length seems quite long - I'm used to passing places which are only big enough for a single vehicle, whereas those on the CA 154 are big enough to "rest areas". As a result, Painted Cave Road looks like it has a nasty disease. 

I can't tell whether `passing_place=left|right` is used consistently as recommended (Painted Cave Road seems to use `side=X`). I suspect a lot of passing places don't have an indicated side?

**Comment by leijurv** [comment]
😹 Yes that's reasonable, it should be shorter. If there is no indicated side I have it drawing on both sides of the road.

**Comment by imagico** [comment]
Just for info - there has been an implementation of this in the AC-Style for some time - see https://github.com/imagico/osm-carto-alternative-colors/blob/master/layers/roads/tc.sql

Be sure to consider the cases where the passing place node is where two road lines meet.

**Comment by leijurv** [comment]
I did that in #5296 but for this one there is an issue: it makes it ambiguous what `side=left` / `side=right` means, if the ways point in different directions. So I'm not sure if there is some way to resolve the ambiguity or if maybe it is best to just not render the passing_place at such a node, as perhaps you could say it is a mapping error to put `side=left` on such an ambiguous node.

**Comment by dch0ph** [comment]
> So I'm not sure if there is some way to resolve the ambiguity or if maybe it is best to just not render the passing_place at such a node, as perhaps you could say it is a mapping error to put `side=left` on such an ambiguous node.

Yes, it would be a mapping error if the directions are inconsistent at a join and a "side" is given. IMO it would make sense to fall back to drawing on both sides.
